### DISCUSSION TITLE
GiveWP Hook after successful donation

### DISCUSSION THREAD

**Discussion by ThemeAZ** [impress-org/givewp#7191] [Developer Q&A]
Hi everyone, 

What is the Hook after successful donation?

I'm trying to add these info to BeaconCRM via API:

How I can get data after  that too?
```
 // Get the payment data
    $payment_data = give_get_payment($payment_id); 
    // Get the donor ID
    $donor_id = $payment_data['user_id'];

    // Extract donor information
    $donor_first_name  = $payment_data['user_info']['first_name'];
    $donor_last_name   = $payment_data['user_info']['last_name'];
    $donor_name        = $donor_first_name . ' ' . $donor_last_name;
    $donor_email       = $payment_data['user_info']['email'];

    // Extract Gift Aid information
    $gift_aid_status   = give_get_meta($payment_id, '_give_giftaid', true);
 
	$donor_object = new Give_Donor( $donor_id );

	$address_one      = $donor_object->get_meta( '_give_gift_aid_card_address', true );
	$address_one      = ! empty( $address_one ) ? $address_one : '';
	$address_two      = $donor_object->get_meta( '_give_gift_aid_card_address_2', true );
	$address_two      = ! empty( $address_two ) ? $address_two : '';
	$gift_aid_city    = $donor_object->get_meta( '_give_gift_aid_card_city', true );
	$gift_aid_city    = ! empty( $gift_aid_city ) ? $gift_aid_city : '';
	$gift_aid_zip     = $donor_object->get_meta( '_give_gift_aid_card_zip', true );
	$gift_aid_zip     = ! empty( $gift_aid_zip ) ? $gift_aid_zip : '';
	$gift_aid_country = $donor_object->get_meta( '_give_gift_aid_country', true );
	$gift_aid_country = ! empty( $gift_aid_country ) ? $gift_aid_country : '';
	$gift_aid_state   = $donor_object->get_meta( '_give_gift_aid_card_state', true );
	$gift_aid_state   = ! empty( $gift_aid_state ) ? $gift_aid_state : '';
	  
```

Thanks.

**Comment by jonwaldstein** [comment]
@ThemeAZ hey!

We have a few different lifecycle hooks that depend on the context of the donation.  The tricky part is that every gateway handles donations differently and will determine for themselves when a donation status is truly successful.

Some hooks we currently have include: ([$donation](https://github.com/impress-org/givewp/blob/develop/src/Donations/Models/Donation.php) = `Give\Donations\Models\Donation;`)
- `do_action('givewp_donation_created', $donation)`
- `do_action('givewp_donation_updated', $donation)`

There is also an older one people have used: `give_update_payment_status`


We also have a webhooks add-on that you might find useful: https://givewp.com/addons/webhooks/

**Reply by ThemeAZ** [reply]
Thank you, I had used "givewp_donation_updated" and it's working.
Only one more thing, how can I Extract donor information? I'd use this code but itn't work.

 ```
// Get the payment data
    $payment_data = give_get_payment($payment_id); 
    // Get the donor ID
    $donor_id = $payment_data['user_id'];

    // Extract donor information
    $donor_first_name  = $payment_data['user_info']['first_name'];
    $donor_last_name   = $payment_data['user_info']['last_name'];
    $donor_name        = $donor_first_name . ' ' . $donor_last_name;
    $donor_email       = $payment_data['user_info']['email'];
```

**Comment by jonwaldstein** [comment]
If you have access to the [$donation](https://github.com/impress-org/givewp/blob/develop/src/Donations/Models/Donation.php) model  (like in the hook below) you can get the [donor model](https://github.com/impress-org/givewp/blob/develop/src/Donors/Models/Donor.php) programmatically:

```
do_action('givewp_donation_updated', function(Give\Donations\Models\Donation $donation) {

        $donor = $donation->donor;
        
        $donorData = [
            'firstName' => $donor->firstName,
            'lastName' => $donor->lastName,
            'email' => $donor->email,
        ];
});
```

Whenever you have access to a `donorId` you can use the Donor model as well:

```
$donor = Give\Donors\Models\Donor::find($donorId);
```

**Reply by ThemeAZ** [reply]
Perfect. Thank you so much.
Current I'd get data(name, email, address,..) from Ajax.
